✨ sign-in page

Chun-Cheng opened the pull request:
Create a sign-in form, and add links to the home page.

After the user submit the form, it'll send sign-in request to the server and print the response to the console.

---

Files inside app/components/ui/ is copied from shadcn/ui.

Chun-Cheng commented:
Checked by linter or not?

Chun-Cheng commented on tsconfig.app.json:
```diff
@@ -14,7 +14,10 @@
+    "paths": {
+      "@/*": ["./src/*"]
+    },
     /* Linting */
```
Comments in json file?

Jeff92316046 replied:
原本就有的

Chun-Cheng replied:
Oops 🫢

Chun-Cheng commented on tsconfig.json:
```diff
@@ -3,5 +3,6 @@
     { "path": "./tsconfig.app.json" },
     { "path": "./tsconfig.node.json" }
-  ]
+  ],
```
Trailing comma is allowed in JSON?

Jeff92316046 replied:
fixed

Chun-Cheng commented on .gitignore:
```diff
@@ -12,6 +12,7 @@ dist
 dist-ssr
 *.local
 
+src/generated/nori/*
```
Should the generated code be ignored?
(We had discussed this issue, but I forgot the conclusion.)

Jeff92316046 replied:
我也忘了
感覺要再討論一遍

Chun-Cheng replied:
Is `src/generated/*` better? The `.gitkeep` has been uploaded, so it should not be affected.

Chun-Cheng replied:
Also, maybe we should use the same directory name in all repos to represent the proto-generated codes, such as `generated` or `proto_gen`?

Chun-Cheng replied:
**proto_generated** 👍

Jeff92316046 replied:
fixed

Jeff92316046 commented:
> Checked by linter or not?

![image](https://github.com/user-attachments/assets/fd9a181d-c619-4919-a0ed-d217c7e59c4b)
生成出來的檔案的error

Chun-Cheng commented:
> > Checked by linter or not?
> 
> ![image](https://private-user-images.githubusercontent.com/83210887/401902129-fd9a181d-c619-4919-a0ed-d217c7e59c4b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY0OTk3NzcsIm5iZiI6MTczNjQ5OTQ3NywicGF0aCI6Ii84MzIxMDg4Ny80MDE5MDIxMjktZmQ5YTE4MWQtYzYxOS00OTE5LWEwZWQtZDIxN2M3ZTU5YzRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEwVDA4NTc1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiODhkYjUxNWZmODQ1MjBiZjEzMzNlNmY2MDc2Yjc4NjliNDQyYzM4ZTYyZWIwMzc5OTg4MmFiYWNkMGNlMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aT1RbZ1EjRPsteEdv2ixkD6J7VIFmFjzYlK8rnge5VI) 生成出來的檔案的error

Ignore it.
(We'd change the eslint setting, but not in this PR.)

Chun-Cheng commented on .gitignore:
```diff
@@ -12,6 +12,8 @@ dist
 dist-ssr
 *.local
 
+src/proto-generated/*
```
Wait, you use hyphen `-` ? Is it allowed to import a module whose name contains hyphens? Consider replacing hyphen `-` with underscore  `_` ?

Sorry for commenting after approved.

Chun-Cheng committed:
🐛 correct path alias

Chun-Cheng committed:
🚧 log-in UI

Chun-Cheng committed:
✨ matrix sdk client

Chun-Cheng committed:
🚧 log-in form

Chun-Cheng committed:
🚸 log-in form value check and loading animation

Chun-Cheng committed:
➕ shadcn/ui alert component

Chun-Cheng committed:
✨ log-in page

Chun-Cheng committed:
🐛 log-in call wrong server

Chun-Cheng committed:
💄 add register link

Chun-Cheng committed:
💄 create home and register page

Chun-Cheng committed:
⚰️ remove unnecessary path alias

Chun-Cheng committed:
💡 add comment

Chun-Cheng committed:
🔥 remove initial and testing page file

Chun-Cheng committed:
♻️ change a to link of react router

reference:
- https://reactrouter.com/start/framework/navigating
- https://how.dev/answers/what-are-the-differences-between-navlink-link-and-a-in-react

Chun-Cheng committed:
♻️ change a to react-router link and remove testing code

Chun-Cheng committed:
🐛 fix log-in page navigate

Chun-Cheng committed:
♻️ modify sync function

Chun-Cheng committed:
♻️ modify debounce helper

Chun-Cheng committed:
🐛 pass refresh token to new SDK client

Chun-Cheng committed:
🔀 Merge branch 'main' into cheng/mes-85-log-in-page

Chun-Cheng committed:
💄 add theme-toggle to pages

Barney9211 reviewed (approved):
看起來挺好的wwwww
